cmd/kops/create_cluster.go

```diff
@@ -472,6 +481,27 @@ func RunCreateCluster(f *util.Factory, out io.Writer, c *CreateClusterOptions) e
 
 	if c.Cloud != "" {
 		cluster.Spec.CloudProvider = c.Cloud
+
+		if c.Cloud == "vsphere" {
+			if cluster.Spec.CloudConfig == nil {
+				cluster.Spec.CloudConfig = &api.CloudConfiguration{}
+			}
+
+			if c.VSphereServer == "" {
+				return fmt.Errorf("vsphere-server is required for vSphere. Set vCenter URL Ex: 10.192.10.30 or myvcenter.io (without https://)")
+			}
+			cluster.Spec.CloudConfig.VSphereServer = c.VSphereServer
```

Comment:
You need spec.cloudconfig to include vSphere information. They will be needed for creating the cloud.config file for cloud provider later. GCE also have their cloudConfig info at the same place. Although they are not accepting user's own configuration, which we need.